app/validators/placeholders/placeholder_validator.py

```diff
@@ -108,7 +108,58 @@ def validate_option_label_from_value_placeholder(self, answer_id):
                 identifier=answer_id,
             )
 
-    def validate_placeholder_transforms(self, transforms):
+    def validate_answer_type_for_transform(
+        self, argument, argument_name, transform_type
+    ):
+        if not (
+            transform_type in ["format_unit", "format_percentage"]
+            and argument_name == "value"
+            and argument.get("value", {}).get("source") == "answers"
+        ):
+            return None
+
+        answer_id = argument["value"]["identifier"]
+        answer_type = self.questionnaire_schema.answers_with_context[answer_id][
+            "answer"
+        ]["type"]
+
+        if answer_type.lower() in transform_type:
+            return None
+
+        self.add_error(
+            error_messages.ANSWER_TYPE_FOR_TRANSFORM_TYPE_INVALID.format(
+                transform=transform_type, answer_type=answer_type
+            ),
+            identifier=answer_id,
+        )
```

Comment:
The error message is not quite right. Currently, you get:
```
"The answer type for 'format_percentage' transform is not of type 'Number'"
```
Should be `"The answer type for 'format_percentage' transform is not of type 'Percentage'"`.

Maybe we should make it more meaningful, so something like:
`"Expected the answer type for 'format_percentage' transform to be type 'Percentage'" but got type 'Number'`.

Reply:
Added the second message as it's much more meaningful.